pkg/alertmanager/alertmanager.go

```diff
@@ -319,12 +325,21 @@ func (am *Alertmanager) Stop() {
 		am.dispatcher.Stop()
 	}
 
+	if service, ok := am.state.(*state); ok {
```

Comment:
I think it's more clean if you cast to the `Service` interface.

```suggestion
	if service, ok := am.state.(services.Service); ok {
```

Reply:
That being said, the next step of refactoring I see (but I would do it in a separate PR) is to require `am.state` to implement the `services.Service` interface. We can wrap `cfg.Peer` into our own `struct` which implements `services.Service` (using `services.NewIdleService()`) and add `services.Service` to `NilPeer` too. This way we keep all implementations consistent.

Thoughts @stevesg @pstibrany ?

Reply:
Seems reasonable. We should consider how to call `Settle` for the non-sharded case too, ideally that would be consistent (the Service wrapper for the Peer wrapper does the settling in starting, as for the replication).

The alternative to the road we've started on, is to have a single "PeerService" which wraps any type of Peer, though that would then need to be extended with some sort of "run" function, so you probably end back at making them all services...